pkg/capacity/capacity_test.go

```diff
@@ -563,23 +561,165 @@ func TestCacheWorkloadOperations(t *testing.T) {
 	}
 }
 
+func TestCapacityUsage(t *testing.T) {
+	capacity := kueue.Capacity{
+		ObjectMeta: metav1.ObjectMeta{Name: "foo"},
+		Spec: kueue.CapacitySpec{
+			RequestableResources: []kueue.Resource{
+				{
+					Name: corev1.ResourceCPU,
+					Flavors: []kueue.ResourceFlavor{
+						{
+							Name: "default",
+							Quota: kueue.Quota{
+								Guaranteed: resource.MustParse("10"),
+								Ceiling:    resource.MustParse("20"),
+							},
+						},
+					},
+				},
+				{
+					Name: "example.com/gpu",
+					Flavors: []kueue.ResourceFlavor{
+						{
+							Name: "model_a",
+							Quota: kueue.Quota{
+								Guaranteed: resource.MustParse("5"),
+								Ceiling:    resource.MustParse("10"),
+							},
+						},
+						{
+							Name: "model_b",
+							Quota: kueue.Quota{
+								Guaranteed: resource.MustParse("5"),
+								Ceiling:    resource.MustParse("10"),
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	workloads := []kueue.QueuedWorkload{
+		{
+			ObjectMeta: metav1.ObjectMeta{Name: "one"},
+			Spec: kueue.QueuedWorkloadSpec{
+				AssignedCapacity: "foo",
+				Pods: []kueue.PodSet{
+					{
+						Count: 1,
+						Spec: utiltesting.PodSpecForRequest(map[corev1.ResourceName]string{
+							corev1.ResourceCPU: "8",
+							"example.com/gpu":  "5",
+						}),
+						AssignedFlavors: map[corev1.ResourceName]string{
+							corev1.ResourceCPU: "default",
+							"example.com/gpu":  "model_a",
+						},
+					},
+				},
+			},
+		},
+		{
+			ObjectMeta: metav1.ObjectMeta{Name: "two"},
+			Spec: kueue.QueuedWorkloadSpec{
+				AssignedCapacity: "foo",
+				Pods: []kueue.PodSet{
+					{
+						Count: 1,
+						Spec: utiltesting.PodSpecForRequest(map[corev1.ResourceName]string{
+							corev1.ResourceCPU: "5",
+							"example.com/gpu":  "6",
+						}),
+						AssignedFlavors: map[corev1.ResourceName]string{
+							corev1.ResourceCPU: "default",
+							"example.com/gpu":  "model_b",
+						},
+					},
+				},
+			},
+		},
+	}
+	cases := map[string]struct {
+		workloads         []kueue.QueuedWorkload
+		wantUsedResources map[corev1.ResourceName]map[string]kueue.Usage
+		wantWorkloads     int
+	}{
+		"single no borrowing": {
```

Comment:
add a test case for a finished workload

Reply:
Good suggestion, but the capacity cache has no knowledge of finished workload. It happens in the event handler of the queued workload. Should we push that down to the cache?
The current implementation is similar to kube-scheduler.

Reply:
Add it in the integration test then.

Reply:
Sg. But I prefer to leave it for #68, which I'm already working on.

Reply:
Ok